src/main/webapp/app/exam/participate/exam-participation.component.ts

```diff
@@ -304,7 +304,7 @@ export class ExamParticipationComponent implements OnInit, OnDestroy, ComponentC
             const exerciseIds = exercises.map((exercise) => exercise.id).filter(Number) as number[];
             this.examParticipationService.setExamExerciseIds(exerciseIds);
             // set endDate with workingTime
-            if (!!this.testRunId || this.testExam) {
+            if (!!this.testRunId || this.studentExam?.exam?.testExam) {
```

Comment:
Is it possible that `this.testExam` also has the wrong value in other places of this component?

Reply:
https://github.com/ls1intum/Artemis/blob/d5cd1f7c87aa1128443cc396cf8a6d8adc14df62/src/main/webapp/app/exam/participate/exam-participation.component.html#L135-L138
Only here in the template, but there is no exam there so this doesn't apply